### DISCUSSION TITLE
Remove fields that do not exist in object y

### DISCUSSION THREAD

**Discussion by rendomnet** [RebeccaStevens/deepmerge-ts#103] [Q&A]
# Suggestion
How this can be done? 

```
const x = {
  record: {
    prop1: "value1",
    prop2: "value2"
  }
};
```

```
const y = {
  record: {
    prop1: "changed"
  }
};

merge(x,y)

//return
{
 record: {
    prop1: "changed"
  }
}
```

**Comment by RebeccaStevens** [comment]
I'm not sure how `y` is supposed to differ from the `result` but essentially this should do what you want:

```ts
import { deepmergeCustom } from "deepmerge-ts";

export const myDeepmerge = deepmergeCustom({
  mergeOthers: (values, utils, meta) => {
    const objectToGetKeysOf = meta.parents[meta.parents.length - 1];
    const keysToKeep = new Set([
      ...Object.keys(objectToGetKeysOf),
      ...Object.getOwnPropertySymbols(objectToGetKeysOf)
    ]);

    if (!keysToKeep.has(meta.key)) {
      return utils.actions.skip;
    }
    return utils.actions.defaultMerge;
  }
});
```

```ts
import { myDeepmerge } from "./path/to/this/function";

const x = {
  record: {
    prop1: "value1",
    prop2: "value2"
  }
};

const y = {
  record: {
    prop1: "changed"
  }
};

const result = myDeepmerge(x, y);
```

**Reply by rendomnet** [reply]
> I'm not sure how y is supposed to differ from the result

Yeah thats right, this is literaly object replacing.
Then how to use this "rule" on the specific object path.

for example "record.deep1.deep2"

```
const x = {
  record: {
    prop1: "value1",
    prop3: "value3",
    deep1: {
     deep2: {
      prop1: "value1",
      prop2: "value2"
     }
    }
  }
};
const y = {
  record: {
    prop1: "valuea",
    prop2: "value2"
    deep1: {
     deep2: {
      prop1: "value1",
     }
    }
  }
};

//return
{
  record: {
    prop1: "valuea",
    prop2: "value2",
    prop3: "value3"
    deep1: {
     deep2: {
      prop1: "value1",
     }
    }
  }
};
```

**Reply by RebeccaStevens** [reply]
Sorry, I'm not exactly sure what you are trying to do. `y` and `return` in that code above are the same and it is also what the code I gave will return.

**Reply by rendomnet** [reply]
> Sorry, I'm not exactly sure what you are trying to do. `y` and `return` in that code above are the same and it is also what the code I gave will return.

edited.

**Reply by rendomnet** [reply]
Gives error
`Cannot read properties of undefined (reading 'defaultMerge')`
https://codesandbox.io/s/nifty-field-2nd9ye?file=/src/example.ts

**Reply by fatihaziz** [reply]
change it to:
```typescript
// ... above codes

    if (!keysToKeep.has(meta.key)) {
      return utils.actions?.skip;
    }
    return utils.actions?.defaultMerge;

// ... below codes
```

so it won't error. there is reason `actions` being `undefined` or `null`

**Reply by RebeccaStevens** [reply]
@rendomnet Sorry, I forgot about this discussion. This issue with the sandbox you linked is that it is using an older version of `deepmerge-ts` (`v3.0.0`). `actions` were introduced in `v4.0.0`.
